storage/innobase/fil/fil0crypt.cc

```diff
@@ -1074,7 +1074,52 @@ static void fil_crypt_write_crypt_data_to_page0(fil_space_t *space) {
   mtr.commit();
 }
 
-bool fil_crypt_exclude_tablespace_from_rotation(fil_space_t *space) {
+bool fil_crypt_exclude_tablespace_from_rotation_temporarily(
+    fil_space_t *space) {
+  // We acquire fil_crypt_threads_mutex to stop encryption threads from
+  // generating crypt_data for tablespaces that they are about to encrypt.
+  // Generating crypt_data is the fist step in encryption process.
+  mutex_enter(&fil_crypt_threads_mutex);
```

Comment:
This seems to be used only on general tabelspaces. Can you add assertion that space is General Tablespace

Reply:
done